### PR TITLE
fix: add proper guard logic, centralized auth checking and redirect

### PR DESCRIPTION
**Pull request by orhanrauf**
A number of things were fixed in this PR:

- the authentication state logic is centralized in the Vuex state
- the router guards now look at single source of truth for auth state
- frontend redirect is reactivity is snappier due to better integration with auth0
- auth state is atomic in single transactions, logIn and logOut, no separate function calls

### CODE REVIEW

File: frontend/src/plugins/1.router/additional-routes.ts [hunk 1]
```diff
@@ -11,13 +11,14 @@ export const redirects: RouteRecordRaw[] = [
     name: 'index',
     redirect: to => {
       const userData = useCookie<Record<string, unknown> | null | undefined>('userData')
-      // const userRole = userData.value?.role
+      
+      // TODO use auth0 scopes
```
Comment (lennertjansen) [inline frontend/src/plugins/1.router/additional-routes.ts:15]: Has this `TODO` been resolved? If not, is it feasible to do so for this PR without speed impairment?

**Review by lennertjansen — APPROVED** [review]
See my one comment on `frontend/src/plugins/1.router/additional-routes.ts`.

Otherwise no remarks. Approved to merge.

File: frontend/src/plugins/1.router/additional-routes.ts (hunk 1 above)
Comment (orhanrauf) [reply frontend/src/plugins/1.router/additional-routes.ts:15]: It's mostly a reminder that this is where we would inject auth0 scopes

Comment (lennertjansen) [reply frontend/src/plugins/1.router/additional-routes.ts:15]: Ok understood. Consider turning it into an issue or adding it to the backlog so we won't accidentally forget about it.

**Pull request merged**